lib/logstash/outputs/opensearch/http_client/manticore_adapter.rb

```diff
@@ -27,13 +38,36 @@ def initialize(logger, options={})
       options[:cookies] = false
 
       @client_params = {:headers => DEFAULT_HEADERS.merge(options[:headers] || {})}
-      
+
+      if options[:auth_type] != nil && options[:auth_type]["type"] == "aws_iam"
```

Comment:
shall we create constant for "aws_iam"?

Reply:
What happens if user set both user name and password and auth type as aws_iam?

Reply:
Sure. will update it